packages/react/test/specs/components/Checkbox/CheckboxBase-test.tsx

```diff
@@ -0,0 +1,185 @@
+    const control = mount(<CheckboxBase />)
+    const inputChecked = control
+      .find('input')
+      .first()
+      .prop('checked')
+    expect(inputChecked).toBe(false)
+    control.setProps({ checked: true })
+    const inputCheckedAfter = control
+      .find('input')
+      .first()
+      .prop('checked')
+    expect(inputCheckedAfter).toBe(true)
+  })
+
+  it('changes state when clicked', () => {
+    const control = mount(<CheckboxBase />)
+    expect(
+      control
+        .find('input')
+        .first()
+        .prop('checked'),
+    ).toBe(false)
+    control.simulate('click')
+    expect(
+      control
+        .find('input')
+        .first()
+        .prop('checked'),
+    ).toBe(true)
+    control.simulate('click')
+    expect(
+      control
+        .find('input')
+        .first()
+        .prop('checked'),
+    ).toBe(false)
+  })
+
+  it('calls onChange when checked state changes', () => {
+    const change = jest.fn()
+    const control = mount(<CheckboxBase onChange={change} />) // let, the only constant is change
+    control.simulate('click')
+    expect(change).toHaveBeenCalled()
+  })
+
+  it('does not change value when onChange prevents default', () => {
+    const change = jest.fn((e: any) => e.preventDefault())
+    const control = mount(<CheckboxBase onChange={change} />)
+    expect(
+      control
+        .find('input')
+        .first()
+        .prop('checked'),
+    ).toBe(false)
+    control.simulate('click')
+    expect(
+      control
+        .find('input')
+        .first()
+        .prop('checked'),
+    ).toBe(false)
+  })
+
+  it('does not call onChange value when onClick prevents default', () => {
+    const click = jest.fn((e: any) => e.preventDefault())
+    const change = jest.fn()
+    const control = mount(<CheckboxBase onClick={click} onChange={change} />)
+    control.simulate('click')
+    expect(change).not.toHaveBeenCalled()
+  })
+
+  it('calls onClick', () => {
+    const click = jest.fn()
+    const control = mount(<CheckboxBase onClick={click} />)
+    control.simulate('click')
+    expect(click).toHaveBeenCalled()
+  })
+
+  it('renders content', () => {
+    const control = mount(
+      <CheckboxBase>
+        <label>foo</label>
+      </CheckboxBase>,
+    )
+    expect(control.find('label').length).toBe(1)
+  })
+
+  it('can render with a different root type', () => {
+    const control = mount(<CheckboxBase slots={{ root: 'span' }} />)
+    expect(control.find('span').length).toBe(1)
+  })
+
+  it('can render with a different input type', () => {
+    const MyInput = () => (
+      <a>
+        <input type="hidden" value="I don't even know why anymore..." />
+      </a>
+    )
+    const control = mount(<CheckboxBase slots={{ input: MyInput }} />)
+    expect(control.find('a').length).toBe(1)
+  })
+
+  it('changes on enter press', () => {
+    const control = mount(<CheckboxBase />)
+    control.simulate('keydown', {
+      keyCode: 13,
+      key: 'Enter',
```

Comment:
Wouldn't the `keyCode` have been enough here?

Reply:
@khmakoto see https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode, it's deprecated.

Only `key` is the valid check moving forward:

https://github.com/levithomason/keyboard-key#why

Reply:
@levithomason sure, what I meant is that having both was redundant, I don't actually have a preference one over the other.